Build with flite 2.1.0 fails

USing Ubuntu 18.10 64bit to build 32bit ffmpeg
The current script uses flite 2.1.0
```
build_libflite() {
  download_and_unpack_file http://www.festvox.org/flite/packed/flite-2.1/flite-2.1-release.tar.bz2
  cd flite-2.1-release
    apply_patch file://$patch_dir/flite-2.1.0_mingw-w64-fixes.patch
    if [[ ! -f main/Makefile.bak ]]; then
      sed -i.bak "s/cp -pd/cp -p/" main/Makefile # friendlier cp for OS X
    fi
    generic_configure
    do_make_and_make_install
  cd ..
}
```

This fails when doing a clean build with this error:
> Installing
mkdir -p /home/c/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/bin
mkdir -p /home/c/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/lib
mkdir -p /home/c/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/include/flite
/usr/bin/install -c -m 644 include/*.h /home/c/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/include/flite
/usr/bin/install: cannot stat '../bin/flite_cmu_us_kal.exe': No such file or directory
/usr/bin/install: cannot stat '../bin/flite_cmu_time_awb.exe': No such file or directory
/usr/bin/install: cannot stat '../bin/flite_cmu_us_kal16.exe': No such file or directory
/usr/bin/install: cannot stat '../bin/flite_cmu_us_awb.exe': No such file or directory
/usr/bin/install: cannot stat '../bin/flite_cmu_us_rms.exe': No such file or directory
/usr/bin/install: cannot stat '../bin/flite_cmu_us_slt.exe': No such file or directory
Makefile:136: recipe for target 'install' failed
make[1]: *** [install] Error 1
Makefile:108: recipe for target 'install' failed
make: *** [install] Error 2

However when using an older scripts flite 2.0
```
build_libflite() {
  download_and_unpack_file http://www.festvox.org/flite/packed/flite-2.0/flite-2.0.0-release.tar.bz2
  cd flite-2.0.0-release
    if [[ ! -f configure.bak ]]; then
      sed -i.bak "s|i386-mingw32-|$cross_prefix|" configure
      #sed -i.bak "/define const/i\#include <windows.h>" tools/find_sts_main.c # Needed for x86_64? Untested.
      sed -i.bak "128,134d" main/Makefile # Library only.
      sed -i.bak "s/cp -pd/cp -p/" main/Makefile # friendlier cp for OS X
    fi
    generic_configure
    #cpu_count=1 # can't handle it :|
    do_make_and_make_install
    #cpu_count=$original_cpu_count
  cd ..
}
```

This builds without any issues

WSL?

On Fri, Jan 21, 2022 at 3:12 PM rboy1 ***@***.***> wrote:

> USing Ubuntu 18.10 64bit to build 32bit ffmpeg
> The current script uses flite 2.1.0
>
> build_libflite() {
>   download_and_unpack_file http://www.festvox.org/flite/packed/flite-2.1/flite-2.1-release.tar.bz2
>   cd flite-2.1-release
>     apply_patch file://$patch_dir/flite-2.1.0_mingw-w64-fixes.patch
>     if [[ ! -f main/Makefile.bak ]]; then
>       sed -i.bak "s/cp -pd/cp -p/" main/Makefile # friendlier cp for OS X
>     fi
>     generic_configure
>     do_make_and_make_install
>   cd ..
> }
>
> This fails when doing a clean build with this error:
>
> Installing
> mkdir -p
> /home/c/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/bin
> mkdir -p
> /home/c/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/lib
> mkdir -p
> /home/c/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/include/flite
> /usr/bin/install -c -m 644 include/*.h
> /home/c/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/include/flite
> /usr/bin/install: cannot stat '../bin/flite_cmu_us_kal.exe': No such file
> or directory
> /usr/bin/install: cannot stat '../bin/flite_cmu_time_awb.exe': No such
> file or directory
> /usr/bin/install: cannot stat '../bin/flite_cmu_us_kal16.exe': No such
> file or directory
> /usr/bin/install: cannot stat '../bin/flite_cmu_us_awb.exe': No such file
> or directory
> /usr/bin/install: cannot stat '../bin/flite_cmu_us_rms.exe': No such file
> or directory
> /usr/bin/install: cannot stat '../bin/flite_cmu_us_slt.exe': No such file
> or directory
> Makefile:136: recipe for target 'install' failed
> make[1]: *** [install] Error 1
> Makefile:108: recipe for target 'install' failed
> make: *** [install] Error 2
>
> However when using an older scripts flite 2.0
>
> build_libflite() {
>   download_and_unpack_file http://www.festvox.org/flite/packed/flite-2.0/flite-2.0.0-release.tar.bz2
>   cd flite-2.0.0-release
>     if [[ ! -f configure.bak ]]; then
>       sed -i.bak "s|i386-mingw32-|$cross_prefix|" configure
>       #sed -i.bak "/define const/i\#include <windows.h>" tools/find_sts_main.c # Needed for x86_64? Untested.
>       sed -i.bak "128,134d" main/Makefile # Library only.
>       sed -i.bak "s/cp -pd/cp -p/" main/Makefile # friendlier cp for OS X
>     fi
>     generic_configure
>     #cpu_count=1 # can't handle it :|
>     do_make_and_make_install
>     #cpu_count=$original_cpu_count
>   cd ..
> }
>
> This builds without any issues
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdp/ffmpeg-windows-build-helpers/issues/604>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAADBUETNJ35UEXD2NNBPCLUXHK5XANCNFSM5MQVLKFQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

--- a/cross_compile_ffmpeg.sh
+++ b/cross_compile_ffmpeg.sh
@@ -1702,15 +1702,12 @@ build_libcaca() {
 }
 
 build_libdecklink() {
-  local url=https://notabug.org/RiCON/decklink-headers.git
-  git ls-remote $url
-  if [ $? -ne 0 ]; then
-    # If NotABug.org server is down , Change to use GitLab.com .
-    # https://gitlab.com/m-ab-s/decklink-headers
-    url=https://gitlab.com/m-ab-s/decklink-headers.git
-  fi
-  do_git_checkout $url
+  # local url=https://gitlab.com/m-ab-s/decklink-headers.git
+  # git ls-remote $url
+  # do_git_checkout $url
+  git clone https://github.com/nanake/decklink-headers.git decklink-headers_git
   cd decklink-headers_git
+    git checkout -b SDK-12.9 origin/SDK/12.9
     do_make_install PREFIX=$mingw_w64_x86_64_prefix
   cd ..
 }

--- a/cross_compile_ffmpeg.sh
+++ b/cross_compile_ffmpeg.sh
@@ -1543,10 +1543,7 @@ build_libsoxr() {
 }
 
 build_libflite() {
-  # download_and_unpack_file http://www.festvox.org/flite/packed/flite-2.1/flite-2.1-release.tar.bz2
-  # original link is not working so using a substitute
-  # from a trusted source
-  download_and_unpack_file http://deb.debian.org/debian/pool/main/f/flite/flite_2.1-release.orig.tar.bz2 flite-2.1-release
+  download_and_unpack_file http://www.festvox.org/flite/packed/flite-2.1/flite-2.1-release.tar.bz2 flite-2.1-release
   cd flite-2.1-release
     apply_patch file://$patch_dir/flite-2.1.0_mingw-w64-fixes.patch
     if [[ ! -f main/Makefile.bak ]]; then

--- a/cross_compile_ffmpeg.sh
+++ b/cross_compile_ffmpeg.sh
@@ -1493,8 +1493,10 @@ build_facebooktransform360() {
 
 build_libbluray() {
   unset JDK_HOME # #268 was causing failure
-  do_git_checkout https://code.videolan.org/videolan/libbluray.git
-  cd libbluray_git
+  # do_git_checkout https://code.videolan.org/videolan/libbluray.git
+  # cd libbluray_git
+  download_and_unpack_file https://download.videolan.org/pub/videolan/libbluray/1.3.4/libbluray-1.3.4.tar.bz2
+  cd libbluray-1.3.4
     if [[ ! -d .git/modules ]]; then
       git submodule update --init --remote # For UDF support [default=enabled], which strangely enough is in another repository.
     else
@@ -1517,10 +1519,11 @@ build_libbluray() {
         sed -i.bak "/WIN32$/,+4d" src/udfread.c # Fix WinXP incompatibility.
       fi
       if [[ ! -f src/udfread-version.h ]]; then
-        generic_meson
+        meson setup builddir --prefix=$mingw_w64_x86_64_prefix --buildtype=release
       fi
     cd ../..
-    generic_meson
+    generic_configure "--disable-examples --disable-bdjava-jar"
+    do_make_and_make_install "CPPFLAGS=\"-Ddec_init=libbr_dec_init\""
   cd ..
 }
 

--- a/cross_compile_ffmpeg.sh
+++ b/cross_compile_ffmpeg.sh
@@ -1517,11 +1517,10 @@ build_libbluray() {
         sed -i.bak "/WIN32$/,+4d" src/udfread.c # Fix WinXP incompatibility.
       fi
       if [[ ! -f src/udfread-version.h ]]; then
-        generic_configure # Generate 'udfread-version.h', or building Libbluray fails otherwise.
+        generic_meson
       fi
     cd ../..
-    generic_configure "--disable-examples --disable-bdjava-jar"
-    do_make_and_make_install "CPPFLAGS=\"-Ddec_init=libbr_dec_init\""
+    generic_meson
   cd ..
 }
 

